x-pack/plugins/endpoint/common/types.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class EndpointAppConstants {
+  static ENDPOINT_INDEX_NAME = 'endpoint-agent*';
+}
+
+export interface EndpointResultList {
+  // the endpoint restricted by the page size
+  endpoints: EndpointData[];
+  // the total number of unique endpoints in the index
+  total: number;
+  // the page size requested
+  request_page_size: number;
+  // the index requested
+  request_page_index: number;
+}
+
+export interface EndpointData {
```

Comment:
@nnamdifrankie Is there a PR that defines this struct in ECS that the Endpoint team agreed upon?

Reply:
@james-elastic 

This our current working agreement. We will need another revision it would seem, now that we have more experience. Once we do this we can make the changes on all sides. 

https://docs.google.com/spreadsheets/d/1ROweh2FrHxEROcXPKWV7bvGEjg2baSA4W5dNxI5yOw8/edit#gid=0

Reply:
@nnamdifrankie Awesome! Can you condense that into a PR that's in the ECS yml format to https://github.com/elastic/endpoint-app-team/pulls  - this is where we're tracking all the ECS data formats that the endpoint will be sending.

Reply:
I think the format should be in source control and easily found by everyone on the teams to make comments about the structure and format.

Reply:
now merged: https://github.com/elastic/endpoint-app-team/pull/118